gwcs/coordinate_frames.py

```diff
@@ -268,23 +481,27 @@ def axis_physical_types(self):
 
         These physical types are the types in frame order, not transform order.
         """
-        return self._axis_physical_types or self._default_axis_physical_types
+        return self._prop.axis_physical_types or self._default_axis_physical_types
 
     @property
-    def _world_axis_object_classes(self):
+    def world_axis_object_classes(self):
         return {f"{at}{i}" if i != 0 else at: (u.Quantity,
                      (),
                      {'unit': unit})
-                for i, (at, unit) in enumerate(zip(self._axes_type, self.unit))}
+                for i, (at, unit) in enumerate(zip(self.axes_type, self.unit))}
 
     @property
-    def _world_axis_object_components(self):
-        return [(f"{at}{i}" if i != 0 else at, 0, 'value') for i, at in enumerate(self._axes_type)]
+    def _native_world_axis_object_components(self):
+        return [(f"{at}{i}" if i != 0 else at, 0, 'value') for i, at in enumerate(self._prop.axes_type)]
 
 
 class CelestialFrame(CoordinateFrame):
     """
-    Celestial Frame Representation
+    Representation of a Celesital coordinate system.
+
+    This class has a native order of longitude then latitude, meaning
+    ``axes_names``, ``unit`` should be lon, lat ordered.  If your transform is
+    in a different order this should be specified with ``axes_order``.
```

Comment:
This turns out to be a lie. axes_names apparently need to be in axes_order order not native order, which I think is a bug.